### PR TITLE
[v11] Suppress logging

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

The new core logger is noisy

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 🔀 Executed flexible sync tests locally if modifying flexible sync
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated

### CODE REVIEW

**Review by papafe — APPROVED** [review]

**Review by elle-j — APPROVED** [review]
Yees! 😄

File: src/js_realm.hpp [hunk 1]
```diff
@@ -598,6 +600,9 @@ inline typename T::Function RealmClass<T>::create_constructor(ContextType ctx)
         realm::disable_sync_to_disk();
     }
 
+    std::shared_ptr<util::NullLogger> logger = std::make_shared<util::NullLogger>();
+    util::Logger::set_default_logger(std::move(logger));
```
Comment (kraenhansen) [inline src/js_realm.hpp:604]: Wouln't it make more sense to set the default log level?

Comment (kneth) [reply src/js_realm.hpp:604]: `NullLogger` is setting it to `off` so no need

Comment (kraenhansen) [reply src/js_realm.hpp:604]: But with this, users can't set the log level to something else and get an output?

Comment (kraenhansen) [reply src/js_realm.hpp:604]: Fair.

**Comment by kneth** [comment]
@kraenhansen The sync logger can still be used when the general logger is silent.

With sync log level = 'all':

Screenshot 2023-04-12 at 10 49 07

With sync log level = 'warn':

Screenshot 2023-04-12 at 10 49 07

File: src/js_realm.hpp (hunk 1 above)
Comment (kneth) [reply src/js_realm.hpp:604]: See my comment below

**Comment by kraenhansen** [comment]
> With sync log level = 'warn'

Those two screenshots look identical to me.
I any way, I'm glad you verified this change can coexist without interfering with the existing behaviour.

**Pull request merged**